sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala

```diff
@@ -413,36 +414,103 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     batchStreamSupported = false,
     streamBatchSupported = false)
 
-  // Left outer joins: *-stream not allowed
+  // Left outer joins: *-stream not allowed with default condition
   testBinaryOperationInStreamingPlan(
     "left outer join",
     _.join(_, joinType = LeftOuter),
-    streamStreamSupported = false,
     batchStreamSupported = false,
-    expectedMsg = "left outer/semi/anti joins")
+    streamStreamSupported = false,
+    expectedMsg = "outer join")
+
+  // Left outer joins: stream-stream allowed with join on watermark attribute
+  assertSupportedInStreamingPlan(
+    s"left outer join with stream-stream relations and join on watermark attribute key",
+    streamRelation.join(streamRelation, joinType = LeftOuter,
+      condition = Some(attributeWithWatermark === attributeWithWatermark)),
+    OutputMode.Append())
+
+  // Left outer joins: stream-stream allowed with range condition yielding state value watermark
+  assertSupportedInStreamingPlan(
+    s"left outer join with stream-stream relations and state value watermark", {
+      val firstRelationWithWatermark = new TestStreamingRelation(attributeWithWatermark)
```

Comment:
can you name this leftRelation/rightRelation, and leftTime/rightTimeWithWatermark? Easier read and reason